File: 250829/트로미노/tromino.java
import java.util.*;

public class Main {

    static int[][][] shapes = new int [][][] {
        {
            {1, 1, 0},
            {1, 0, 0},
            {0, 0, 0}
        },
        {
            {1, 1, 0},
            {0, 1, 0},
            {0, 0, 0}
        },
        {
            {0, 1, 0},
            {1, 1, 0},
            {0, 0, 0}
        },
        {
            {1, 0, 0},
            {1, 1, 0},
            {0, 0, 0}
        },
        {
            {1, 1, 1},
            {0, 0, 0},
            {0, 0, 0}
        },
        {
            {1, 0, 0},
            {1, 0, 0},
            {1, 0, 0}
        }
    };

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        int[][] grid = new int[n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                grid[i][j] = sc.nextInt();
            }
        }

        // Please write your code here.
        int max = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {

                for (int s = 0; s < 6; s++) {
                    int sum = 0;
                    boolean isRight = true;

                    for (int x = 0; x < 3; x++) {
                        for (int y = 0; y < 3; y++) {
                            if (shapes[s][x][y] == 0) continue;
                            
                            // if (i + x < 0 || j + y < 0) {
                            //     System.out.println("hi: " + i + " " + x + " " 
                            //     + j + ' ' + y);
                            //     return;
                            // }
                            if (i + x < n && j + y < m) {
                                sum += grid[i + x][j + y];
                            } else {
                                isRight = false;
                            }
                        }
                    }
                    if (isRight)
                        max = Math.max(max, sum);
                }             
            }
        }

        System.out.println(max);

    }
}

File: 260316/트로미노/tromino.py
n, m = map(int, input().split())
grid = [list(map(int, input().split())) for _ in range(n)]

# Please write your code here.
def log(msg):
    # print(msg)
    pass

shapes = [
    [(0,0), (0,1), (0,2)],
    [(0,0), (1,0), (2,0)],

    [(0,0), (1,0), (1,1)],
    [(0,1), (1,0), (1,1)],
    [(0,0), (0,1), (1,1)],
    [(0,0), (1,0), (0,1)],
]

max_sum = 0

for r in range(n):
    for c in range(m):
        for shape in shapes:
            sum = 0
            for dr, dc in shape:
                rr = r + dr
                cc = c + dc
                if 0 <= rr and rr < n and 0 <= cc and cc < m:
                    sum += grid[rr][cc]
                else:
                    sum = -1
                    break
            
            if max_sum < sum:
                max_sum = sum
                log(f'{r} {c} {max_sum}')

print(max_sum)